src/components/PresenceMap/component.js

```diff
@@ -0,0 +1,194 @@
+        const ctx = canvas.getContext('2d');
+        canvas.width = 40;
+        canvas.height = 50;
+        const cache = img;
+        const imageWidth = cache.width;
+        const imageHeight = cache.height;
+        ctx.save();
+        ctx.clearRect(0, 0, canvas.width, canvas.height);
+        ctx.translate(canvas.width / 2, canvas.height / 2);
+        ctx.rotate((Math.PI / 180) * heading);
+        ctx.translate(-(canvas.width / 2), -(canvas.height / 2));
+        const xCenter = canvas.width / 2 - imageWidth / 2;
+        const yCenter = canvas.height / 2 - imageHeight / 2;
+        ctx.drawImage(img, xCenter, yCenter, imageWidth, imageHeight);
+        ctx.restore();
+
+        return canvas.toDataURL();
+    };
+
+    const initMap = () => {
+        const mapOptions = {
+            zoom: zoom === 'auto' ? MAX_ZOOM : zoom,
+            center: center !== 'auto' ? center : null,
+            disableDefaultUI: true,
+        };
+        // eslint-disable-next-line
+        const map = new window.google.maps.Map(mapContainer.current, mapOptions);
+        const markersCopy = [];
+
+        objects.forEach(marker => {
+            const heading = marker.heading ? marker.heading : 0;
+
+            const markerOptions = {
+                position: marker.position,
+                map,
+                zIndex: Math.round(marker.position.lat * 100000),
+            };
+
+            let promImage = null;
+
+            if (marker.image) {
+                promImage = new Promise((resolve, reject) => {
+                    const result = getBase64Image(marker.image, heading);
+                    result
+                        .then(img => {
+                            markerOptions.icon = img;
+                            resolve(true);
+                        })
+                        .catch(error => {
+                            reject(`Error creating new image. Details: ${error}`);
+                        });
+                });
+            } else {
+                promImage = new Promise(resolve => resolve(true));
+            }
+
+            promImage
+                .then(() => {
+                    const markerMap = new window.google.maps.Marker(markerOptions);
+                    markersCopy.push(markerMap);
+
+                    if (center === 'auto') {
+                        const bounds = new window.google.maps.LatLngBounds();
+                        markersCopy.forEach(markerObj => bounds.extend(markerObj.getPosition()));
+                        map.setCenter(bounds.getCenter());
+                        map.fitBounds(bounds);
+                    }
+
+                    setMap(map);
+                })
+                .catch(error => {
+                    // eslint-disable-next-line no-console
+                    console.log(`Error. Details: ${error}`);
+                });
+        });
+    };
+
+    return (
+        <StyledContainer style={style} ref={container}>
+            <StyledMapContainer
```

Comment:
I think we are adding an unnecessary wrapper to the component. 
why do we need to have two wrappers with practically the same styles?